### PR TITLE
WIP: [Link] component

### LINKED ISSUE #149: [Link] component
As a consumer using react router, i want my links to go through my routing and have all the CA styles. Additionally, for people that know the difference between the usage of link and button, we should provide a solution for a link, if not a link component.

AC:
1. Provide a consistently styled link.
2. Make sure this link works with React Router.
3. Show your work

Other considerations:
- Warn if using href on Button or onClick on Link?
- Do and Don'ts to reflect usage as well.
- Should this be its own component?
    - Or should Button take an href prop?
- Does Button need a way to display without padding?

### PR DESCRIPTION
**Pull request by riley**
### Description
Added a Link component. This really just a wrapper around an `<a>` since we don't want to re-invent the wheel re: keyboard events, a11y, etc.

closes #149 

### Motivation and context
We decided that Link needs to be a separate component than Button, since a button is a form element, while an anchor is not.

### How to test
1. `npm run build`
2. navigate to Site > Link
3. Play with the links and make sure all the normal props are passed thru
4. It's pretty much just an anchor tag

### Types of changes
- New feature

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support)
* [x] Automated tests written and passing
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered
* [x] Documentation created or updated

### How does this PR make you feel?
hypertext is awesome

### CODE REVIEW

File: packages/link/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+## Usage
+
+```jsx
+import React from 'react';
+import Link from '@mineral-ui/link';
+
+export default function MyComponent() {
+  return (
+    <div>
+      <Link href="https://yourawesomesite.com">
```
Comment (kylegach) [inline packages/link/README.md]: We should link to `http://mineral-ui.com`.

File: packages/link/src/Link.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+type Props = {
+  /** Content of the links. Can be anything, but see LinkBlock, LinkImage, and LinkTitle. */
+  children: MnrlReactNode,
+  /** URI tied to the Link */
+  href: string
+};
+
+const Root = createStyledComponent(
+  'a',
+  (props, baseTheme) => {
```
Comment (kylegach) [inline packages/link/src/Link.js]: Lines 29 – 31 should all be on one line. Not sure why Prettier didn't catch that.

File: packages/link/src/Link.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  return {
+    fontSize: theme.Link_fontSize
+  };
+});
+
+/**
+ * Link component
+ */
+export default function Link(props: Props) {
+  const rootProps = {
```
Comment (kylegach) [inline packages/link/src/Link.js]: You need to pull `children` out so that you're not applying it to `Root`:

```
const rootProps = {
    children,
    ...restProps
};
return <Root {...restProps}>...
```

File: packages/link/src/__demo__/examples/index.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+import linkExamples from './link';
+
+const examples = [].concat(linkExamples);
```
Comment (kylegach) [inline packages/link/src/__demo__/examples/index.js]: This pattern (and the entire file) is only necessary for packages with subcomponents.

File: packages/link/src/__demo__/examples/link/arbitraryChildren.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+/* @flow */
+import { mineralTheme } from '@mineral-ui/component-utils';
+import Link from '../../../Link';
+
+export default {
+  title: 'Arbitrary children',
+  // $FlowFixMe
+  backgroundColor: mineralTheme.color_gray_10,
```
Comment (kylegach) [inline packages/link/src/__demo__/examples/link/arbitraryChildren.js]: Probably not necessary.

File: packages/link/src/__demo__/examples/link/states.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    marginRight: '0.5rem'
+  }
+});
+
+export default {
+  title: 'States',
+  hideSource: true,
+  scope: { Link, DemoLayout, simulate },
+  source: `<DemoLayout>
+  <Link href="http://example.com">Don't Panic</Link>
```
Comment (kylegach) [inline packages/link/src/__demo__/examples/link/states.js]: Love it. :)

File: packages/link/src/__tests__/Link.spec.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+describe('Link', () => {
+  let link;
+  it('renders', () => {
+    link = renderLink({ href: 'http://example.com' }, 'Children');
+
+    expect(link.exists()).toEqual(true);
+  });
+
+  describe('onClick', () => {
```
Comment (kylegach) [inline packages/link/src/__tests__/Link.spec.js]: This is just default browser behavior. Since we're not doing anything with the `onClick` prop in the component (it just gets passed through, because it's a standard DOM attribute for `a`), we probably don't need tests for it.

File: packages/link/src/linkTheme.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+export default function linkTheme(baseTheme: Object) {
```
Comment (kylegach) [inline packages/link/src/linkTheme.js]: I think we decided to keep these in the component file, unless there's plenty of reason (like subcomponents) to split it out.

File: packages/link/README.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
+# Card
```
Comment (coldpour) [inline packages/link/README.md]: needs updates

File: packages/link/README.md (hunk 1 above)
Comment (coldpour) [inline packages/link/README.md]: yup

File: packages/link/package.json [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  "devDependencies": {
+    "@mineral-ui/component-utils": "^0.3.0"
+  },
+  "peerDependencies": {
+    "@mineral-ui/component-utils": "^0.3.0",
+    "glamor": "^2.0.0",
+    "glamorous": "^3.0.0",
+    "react": "^15.0.0",
+    "react-dom": "^15.0.0",
+    "react-router-dom": "^4.1.2"
```
Comment (coldpour) [inline packages/link/package.json]: @brentertz is this cool? hot or not?

File: packages/link/src/Link.js [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+import React from 'react';
+import { Link as RouterLink } from 'react-router-dom';
+import { createStyledComponent } from '@mineral-ui/component-utils';
+
+type Props = {
+  /** Content of the link. Can be anything */
+  children: MnrlReactNode,
+  /** URI tied to the Link */
+  to: string
+};
```
Comment (coldpour) [inline packages/link/src/Link.js]: something is messed up with our props cause the proptable is not rendering in the site...

File: packages/link/src/__demo__/examples/link.js [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+/* @flow */
+import { mineralTheme } from '@mineral-ui/component-utils';
+import Link from '../../Link';
+
+export default {
+  title: 'Basic link',
+  description: 'an anchor tag',
+  scope: { Link },
+  source: `<Link to="https://example.com">Make it so</Link>`
+};
```
Comment (coldpour) [inline packages/link/src/__demo__/examples/link.js]: need a prose example

File: packages/site/src/components/__tests__/__snapshots__/App.spec.js.snap [hunk 13]
```diff
@@ -912,6 +912,78 @@ exports[`App renders correctly 1`] = `
...
+                  "World": [Function],
+                },
+                "source": "<World text=\\"Felix\\" />",
+                "title": "Custom Text",
+              },
+            ],
+            "slug": "world",
+            "title": "World",
+          },
+>>>>>>> WIP: linting
```
Comment (coldpour) [inline packages/site/src/components/__tests__/__snapshots__/App.spec.js.snap]: oops

File: packages/link/CHANGELOG.md [hunk 14]
```diff
@@ -0,0 +1,23 @@
+# Change Log
```
Comment (brentertz) [inline packages/link/CHANGELOG.md]: Please delete all content within this file.

**Comment by brentertz** [comment]
This PR has a `packages/link/package-lock.json` file which must be deleted.  We are not using lock files.  If it reappears, it is a smell that you have some mineral dependency versioning problems on your branch.  I can help.

File: packages/link/src/__demo__/examples/states.js [hunk 15]
```diff
@@ -13,27 +13,26 @@
...
-  forwardProps: ['to']
-});
-
-export default createStyledComponent('a', styles);
+export default {
+  title: 'States',
+  hideSource: true,
+  scope: { Link, DemoLayout, simulate },
+  source: `<DemoLayout>
+  <Link to="http://example.com">Don't Panic</Link>
```
Comment (brentertz) [inline packages/link/src/__demo__/examples/states.js]: As Kyle suggested, linking to mineral-ui would be better.

File: packages/link/src/Link.js [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+  const units = measurement.replace(value, '');
+  return `${func(value)}${units}`;
+}
+
+const Root = createStyledComponent(RouterLink,
+  (props, theme) => {
+    return {
+      color: theme.color_link,
+      fontWeight: theme.Link_fontWeight || theme.fontWeight_regular,
+      fontSize: theme.Link_fontSize || transform(theme.fontSize_base, v => v*2),
```
Comment (brentertz) [inline packages/link/src/Link.js]: I'd like to get @kylegach to review these styles.

File: packages/site/src/components/Footer.js [hunk 17]
```diff
@@ -17,7 +17,7 @@
 /* @flow */
 import React from 'react';
 import { createStyledComponent } from '@mineral-ui/component-utils';
-import Link from './Link';
+import Link from '@mineral-ui/link';
```
Comment (brentertz) [inline packages/site/src/components/Footer.js]: Good thinking to update the site to use the new Link component.  Let's just try to keep this stuff as a separate commit.

File: packages/link/src/__demo__/examples/arbitraryChildren.js [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+
+/* @flow */
+import { mineralTheme } from '@mineral-ui/component-utils';
+import Link from '../../Link';
+
+export default {
+  title: 'Arbitrary children',
+  description: 'A Link will render any children.',
+  scope: { Link },
+  source: `<Link to="http://example.com">
```
Comment (brentertz) [inline packages/link/src/__demo__/examples/arbitraryChildren.js]: This link doesn't actually work.  I click it and it changes the url to `http://0.0.0.0:8080/http://example.com` but it does not navigate.

File: packages/link/src/Link.js [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+  {
+    displayName: 'Link',
+    includeStyleReset: true
+  }
+);
+
+/**
+ * Link component
+ */
+export default function Link({children, ...otherProps}: Props) {
```
Comment (brentertz) [inline packages/link/src/Link.js]: Our convention is to use the name `restProps` rather than `otherProps`.

File: packages/link/src/Link.js [hunk 20]
```diff
@@ -0,0 +1,66 @@
...
+    };
+  },
+  {
+    displayName: 'Link',
+    includeStyleReset: true
+  }
+);
+
+/**
+ * Link component
```
Comment (brentertz) [inline packages/link/src/Link.js]: Perhaps we can find a better way to describe this component, rather than simply Link component.  Check here for some inspiration.  https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a

File: packages/link/src/__demo__/examples/arbitraryChildren.js [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
+  <img src="https://www.fillmurray.com/100/300" />
```
Comment (brentertz) [inline packages/link/src/__demo__/examples/arbitraryChildren.js]: Can this image be smaller?

File: packages/link/package.json (hunk 10 above)
Comment (brentertz) [inline packages/link/package.json]: Probably not.

**Comment by brentertz** [comment]
Any thoughts to what happens if users add an `onClick` handler?  Does it get a `role="button"`, do we warn them, in dev, that they should use a Button instead?  Is it a documentation thing?

File: packages/link/src/Link.js (hunk 16 above)
Comment (brentertz) [inline packages/link/src/Link.js]: I don't see a theme at all.  This component should have a theme of it's own.

File: packages/link/src/__demo__/examples/link.js (hunk 12 above)
Comment (brentertz) [inline packages/link/src/__demo__/examples/link.js]: This example also does not work as expected.  It goes to `http://0.0.0.0:8080/https://example.com`

File: packages/link/src/Link.js [hunk 22]
```diff
@@ -0,0 +1,66 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+import React from 'react';
+import { Link as RouterLink } from 'react-router-dom';
```
Comment (brentertz) [inline packages/link/src/Link.js]: I don't think that we want to directly tie this package to react-router.  

One solution might be to have this component use a regular `a` tag, but allow the base component to be specified via a prop.  Then users could pass in a react-router `Link` or `NavLink` and that would get custom styled.  As long as all props, including the `to` prop,  are passed through to the base component, this should work. This would avoid any direct dependencies.  We would of course need some examples, demonstrating this functionality.

File: packages/link/src/Link.js [hunk 23]
```diff
@@ -0,0 +1,66 @@
...
+import { createStyledComponent } from '@mineral-ui/component-utils';
+
+type Props = {
+  /** Content of the link. Can be anything */
+  children: MnrlReactNode,
+  /** URI tied to the Link */
+  to: string
```
Comment (brentertz) [inline packages/link/src/Link.js]: We should consider what other props to surface.  e.g. `href`, `title`, `target`, etc.

Comment (brentertz) [inline packages/link/src/Link.js]: react-router allows the `to` prop to be a string or an object.  My recommendation is likely to not include that prop here, but rather build a basic styled Link component that uses props based on standard HTML attributes.  If users choose to change the base component to a react-router Link, as described elsewhere, those props can be passed through via restProps, though they should not need declared.

**Comment by kylegach** [comment]
Re: https://github.com/mineral-ui/mineral-ui/pull/211#issuecomment-319367427

I read a fascinating discussion about this decision the other day: https://github.com/alphagov/govuk_elements/pull/272

File: packages/site/src/components/__tests__/__snapshots__/App.spec.js.snap (hunk 13 above)
Comment (coldpour) [inline packages/site/src/components/__tests__/__snapshots__/App.spec.js.snap]: fixed

File: packages/link/src/Link.js (hunk 11 above)
Comment (coldpour) [inline packages/link/src/Link.js]: fixed

File: packages/link/README.md [hunk 24]
```diff
@@ -0,0 +1,32 @@
...
+
+```jsx
+import React from 'react';
+import Link from '@mineral-ui/link';
+
+export default function MyComponent() {
+  return (
+    <div>
+      <Link href="https://mineral-ui.com">
+        Best Site in the universe
```
Comment (kylegach) [inline packages/link/README.md]: Not sure this is a good look.

File: packages/link/package.json (hunk 10 above)
Comment (kylegach) [inline packages/link/package.json]: Is `react-router-dom` still a peer dependency because of the demo example?

File: packages/link/src/Link.js [hunk 25]
```diff
@@ -0,0 +1,65 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+import React from 'react';
+import { createStyledComponent } from '@mineral-ui/component-utils';
+
+type Props = {
+  /** Content of the link. Can be anything */
```
Comment (kylegach) [inline packages/link/src/Link.js]: Please match the style of the other components (e.g. no period). I also don't think "Can be anything" is necessary, especially since you have an example demonstrating so.

File: packages/link/src/Link.js [hunk 26]
```diff
@@ -0,0 +1,65 @@
...
+  children: MnrlReactNode,
+  /** Link renders an <a> by default but you can specify a different tag or component as the root (e.g. a React Router Link).*/
```
Comment (kylegach) [inline packages/link/src/Link.js]: We have a Default column; no need to also spell it out in the Description.

File: packages/link/src/Link.js [hunk 27]
```diff
@@ -0,0 +1,65 @@
...
+  root?: MnrlReactNode
```
Comment (kylegach) [inline packages/link/src/Link.js]: How do you feel about `element`, instead? `root` doesn't seem very clear. Could just be me, though.

File: packages/link/src/Link.js [hunk 28]
```diff
@@ -0,0 +1,65 @@
...
+};
+
+/**
+ * Link component
+ */
+export default function Link({ children, root = 'a', ...otherProps }: Props) {
```
Comment (kylegach) [inline packages/link/src/Link.js]: `otherProps` -> `restProps`, per the pattern used in other components.

File: packages/link/src/Link.js [hunk 29]
```diff
@@ -0,0 +1,65 @@
...
+  if (otherProps.onClick) {
+    console.error(
```
Comment (kylegach) [inline packages/link/src/Link.js]: Will a `console.error` potentially stop a build in a consuming app? If so, is there _no_ legitimate reason to use an `onClick` on a link? Seems like there could be (tracking purposes come to mind).

File: packages/link/src/Link.js [hunk 30]
```diff
@@ -0,0 +1,65 @@
...
+      'You put an onClick on a Link. Maybe you want a Button instead?'
+    );
+  }
+  const Root = createStyledComponent(
+    root,
+    (props, theme) => {
```
Comment (kylegach) [inline packages/link/src/Link.js]: To follow the pattern in our other components, this function should be:

```js
(props, baseTheme) => {
  const theme = {
    Link_color: baseTheme.color_link,
    Link_color_active: baseTheme.color_link_active,
    Link_color_focus: baseTheme.color_link_focus,
    Link_color_hover: baseTheme.color_link_hover,
    Link_fontWeight: baseTheme.fontWeight_regular,
    Link_fontSize: `${2 * baseTheme.fontSize_base}px`

    ...baseTheme
  };

  return {
    color: theme.Link_color,
    fontWeight: theme.Link_fontWeight,
    fontSize: theme.Link_fontSize,

    '&:hover': {
      color: theme.Link_color_hover
    },
    '&:focus': {
      color: theme.Link_color_focus
    },
    // `:active` must be last, to follow LVHFA order:
    // https://developer.mozilla.org/en-US/docs/Web/CSS/:active
    '&:active': {
      color: theme.Link_color_active
    }
  };
}
```

File: packages/link/src/Link.js [hunk 31]
```diff
@@ -0,0 +1,65 @@
...
+      return {
+        color: theme.color_link,
+        fontWeight: theme.Link_fontWeight || theme.fontWeight_regular,
+        fontSize: theme.Link_fontSize || `${2 * theme.fontSize_base}px`,
+
+        '&:hover': {
+          color: theme.color_link_hover
+        },
+        '&:focus': {
```
Comment (kylegach) [inline packages/link/src/Link.js]: This focus style is not even remotely discernible. Please follow the `boxShadow` design used in Button (but with no padding).

File: packages/link/src/__demo__/examples/reactRouter.js [hunk 32]
```diff
@@ -0,0 +1,45 @@
...
+  Link as ReactRouterLink,
+  Route,
+  Switch
+} from 'react-router-dom';
+
+export default {
+  title: 'Using React Router',
+  description: `In this example, we import { Link as ReactRouterLink } from 'react-router-dom' so that we can get client-side routing with our styles.`,
+  scope: { Link, ReactRouterLink, BrowserRouter, Switch, Route },
+  source: `<BrowserRouter>
```
Comment (kylegach) [inline packages/link/src/__demo__/examples/reactRouter.js]: _Not relevant to this PR_: We thought of the need for examples with a hidden source, but didn't think of examples that need to hide the preview, like this one.

File: packages/link/src/__demo__/examples/states.js [hunk 33]
```diff
@@ -13,27 +13,26 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import darken from 'polished/lib/color/darken';
```
Comment (kylegach) [inline packages/link/src/__demo__/examples/states.js]: Should we be concerned that git apparently thinks this file is a renamed `packages/site/src/components/Link.js`?

File: packages/site/src/components/Footer.js (hunk 17 above)
Comment (kylegach) [inline packages/site/src/components/Footer.js]: ⬆️ Reminder

File: packages/site/src/components/__tests__/__snapshots__/App.spec.js.snap [hunk 34]
```diff
@@ -912,6 +912,77 @@ exports[`App renders correctly 1`] = `
...
+          <Link {...simulate('focus')} href=\\"http://example.com\\">Focus</Link>
+          <Link {...simulate('active')} href=\\"http://example.com\\">Active</Link>
+        </DemoLayout>",
+                "title": "States",
+              },
+            ],
+            "slug": "link",
+            "title": "Link",
+          },
+          "world": Object {
```
Comment (kylegach) [inline packages/site/src/components/__tests__/__snapshots__/App.spec.js.snap]: I think you need to update this snapshot after rebasing from master. `Hello`, `World`, and `HelloWorld` no longer exist.

**Comment by riley** [comment]
@brentertz The solution we have right now is to have the component print a `console.error` with a warning (`console.warn` doesn't have a stack trace).

Another story we'll have to definitely add is a way to style the Link as a Button, since sometimes implementers are going to want a call to action "button" that takes the user to another page.

File: packages/link/src/Link.js (hunk 27 above)
Comment (riley) [inline packages/link/src/Link.js]: maybe `rootElement`?

File: packages/link/src/Link.js (hunk 29 above)
Comment (riley) [inline packages/link/src/Link.js]: It won't act like a thrown error if I'm understanding your question. We used `console.error` so the user could get a stack trace to get a hint where the component was implemented.

File: packages/link/README.md (hunk 24 above)
Comment (riley) [inline packages/link/README.md]: whoops

File: packages/link/package.json (hunk 10 above)
Comment (coldpour) [inline packages/link/package.json]: yes

Comment (coldpour) [inline packages/link/package.json]: er sorry, that makes it a devDependency. it does not need to be installed by a consumer

File: packages/link/src/__demo__/examples/arbitraryChildren.js [hunk 35]
```diff
@@ -0,0 +1,27 @@
...
+
+/* @flow */
+import Link from '../../Link';
+
+export default {
+  title: 'Arbitrary children',
+  description: 'A Link will render any children.',
+  scope: { Link },
+  source: `<Link href="http://example.com">
+  <img src="https://www.fillmurray.com/100/100" />
```
Comment (brentertz) [inline packages/link/src/__demo__/examples/arbitraryChildren.js]: Image should have an alt attribute

File: packages/link/src/__demo__/examples/states.js [hunk 36]
```diff
@@ -13,27 +13,26 @@
...
-  forwardProps: ['to']
-});
-
-export default createStyledComponent('a', styles);
+export default {
+  title: 'States',
+  hideSource: true,
+  scope: { Link, DemoLayout, simulate },
+  source: `<DemoLayout>
+  <Link href="http://example.com">Don't Panic</Link>
```
Comment (brentertz) [inline packages/link/src/__demo__/examples/states.js]: We should just point all of these links to http://mineral-ui.com

File: packages/link/src/Link.js (hunk 30 above)
Comment (brentertz) [inline packages/link/src/Link.js]: Agreed.  Let's follow the pattern that exists in other components.

File: packages/link/src/Link.js [hunk 37]
```diff
@@ -0,0 +1,65 @@
...
+  children: MnrlReactNode,
+  /** Link renders an <a> by default but you can specify a different tag or component as the root (e.g. a React Router Link).*/
+  root?: MnrlReactNode
+};
+
+/**
+ * Link component
+ */
+export default function Link({ children, root = 'a', ...restProps }: Props) {
+  if (restProps.onClick) {
```
Comment (brentertz) [inline packages/link/src/Link.js]: I like the intention of enforcing the distinction between buttons and link, but I think logging an error or even a warning can potentially disrupt developer workflows, and we would never want to do this in production.  I think that we'll need to solve this via documentation once we begin writing usage guidelines.

File: packages/link/src/__tests__/Link.spec.js [hunk 38]
```diff
@@ -0,0 +1,37 @@
...
+import Link from '../Link';
+import examples from '../__demo__/examples';
+import testDemoExamples from '../../../../utils/test/testDemoExamples';
+
+function renderLink(props, children) {
+  return shallow(<Link {...props}>{children}</Link>);
+}
+
+describe('Link', () => {
+  let link;
```
Comment (brentertz) [inline packages/link/src/__tests__/Link.spec.js]: No need for this `let` . It can be a `const link` inside the `it` block.

File: packages/link/package.json [hunk 39]
```diff
@@ -0,0 +1,41 @@
...
+    "flow": "$npm_package_config_script_path/flow.sh",
+    "jest": "$npm_package_config_script_path/jest.sh",
+    "lint": "$npm_package_config_script_path/lint.sh",
+    "pack-ls": "$npm_package_config_script_path/pack-ls.sh",
+    "start": "DEMO_LIST_PATH=\"link\" $npm_package_config_script_path/start.sh",
+    "test": "$npm_package_config_script_path/test.sh"
+  },
+  "devDependencies": {
+    "@mineral-ui/component-utils": "^1.0.0",
+    "react-router-dom": "^4.1.2"
```
Comment (brentertz) [inline packages/link/package.json]: This only needs to appear in the project's package.json

File: packages/link/src/__demo__/examples/link.js [hunk 40]
```diff
@@ -0,0 +1,27 @@
...
+
+/* @flow */
+import Link from '../../Link';
+
+export default {
+  title: 'Basic link',
+  description: 'Links can be used in prose to create hypertext.',
+  scope: { Link },
+  source: `<p>
+Sensors indicate human life forms 30 meters below the planet surface. Stellar flares are increasing in magnitude and frequency. Set course for <Link target="_blank" href="http://memory-alpha.wikia.com/wiki/Rhomboid_Dronegar_sector_006">Rhomboid Dronegar 006</Link>, warp seven. There is no evidence of an advanced communication network. Total guidance system failure, with less than 24 hours reserve power. Shield effectiveness has been reduced 12 percent. We have covered the area in a spherical pattern which a ship without warp drive could cross in the given time.
```
Comment (brentertz) [inline packages/link/src/__demo__/examples/link.js]: Please just reuse the lipsum that is used elsewhere on the site.  We want to try to keep the language standard/professional until we have defined a voice of our own.

**Comment by brentertz** [comment]
@kylegach Can you please run the focus styles by the design team.  I personally think that they are a bit odd/unexpected.

**Comment by brentertz** [comment]
I am still feeling a bit wishy-washy on the prop named, "element".  Can we please revisit it one last time?

**Comment by brentertz** [comment]
Re: https://github.com/mineral-ui/mineral-ui/pull/211#issuecomment-320246081 — I updated it to simply use the browser’s focus styles, for now.

**Comment by kylegach** [comment]
Re: https://github.com/mineral-ui/mineral-ui/pull/211#issuecomment-320246081 — I updated it to simply use the browser’s focus styles, until we receive a design spec.
